src/main/java/edu/elka/peakadvisor/calculator/Calculator.java

```diff
@@ -20,7 +20,7 @@
     public Calculator () {
     }
 
-    public ArrayList<Rate> predictRates (ArrayList<Rate> rates, long begin, long end) {
+    public ArrayList<Rate> predictRatesLinear (ArrayList<Rate> rates, long begin, long end) {
         ArrayList<Rate> results = new ArrayList<>();
```

Comment:
referencja do listy powinna być typem interfejsowym tj.
"List<Rate> list = new ArrayList<Rate>();" jak możesz to podmień to i typ zwracany z ArrayList na List